sklearn/utils/estimator_checks.py

```diff
@@ -726,31 +726,51 @@ def check_sample_weights_invariance(name, estimator_orig):
 
         estimator1 = clone(estimator_orig)
         estimator2 = clone(estimator_orig)
+        estimator3 = clone(estimator_orig)
         set_random_state(estimator1, random_state=0)
         set_random_state(estimator2, random_state=0)
+        set_random_state(estimator3, random_state=0)
 
         X = np.array([[1, 3], [1, 3], [1, 3], [1, 3],
                       [2, 1], [2, 1], [2, 1], [2, 1],
                       [3, 3], [3, 3], [3, 3], [3, 3],
                       [4, 1], [4, 1], [4, 1], [4, 1]], dtype=np.dtype('float'))
         y = np.array([1, 1, 1, 1, 2, 2, 2, 2,
                       1, 1, 1, 1, 2, 2, 2, 2], dtype=np.dtype('int'))
+
+        X2 = np.vstack([X, X])
+        y2 = np.hstack([y, 3 - y])
         y = _enforce_estimator_tags_y(estimator1, y)
+        y2 = _enforce_estimator_tags_y(estimator3, y2)
+        weights = np.ones(shape=len(y) * 2)
+        weights[len(y):] = 0
+        X2, y2, weights = shuffle(X2, y2, weights, random_state=0)
```

Comment:
Shuffling the data might have an impact, isn't it.
I was checking and OneClassSVM will give the same result if you are not shuffling the data.

Reply:
Yes indeed. Your test is better wrt to that, I think. I didn't want things to be sorted by weight (CV estimators will fail) but I guess I would need to ensure the order of the non-zero-weight samples stays the same (which is I think what you did?)

Reply:
Also the default `tol` is too high to compare the decision function. I just tried with `SVC` with a `tol=1e-12` then the decision function will be really close.